debugger/src/debugger.rs

```diff
@@ -607,35 +740,47 @@ impl Debugger {
                 match debug_adapter.last_known_status {
                     CoreStatus::Unknown => Ok(DebuggerStatus::ContinueSession), // Don't do anything until we know VSCode's startup sequence is complete, and changes this to either Halted or Running.
                     CoreStatus::Halted(_) => {
-                        // No need to poll the target if we know it is halted and waiting for us to do something.
+                        // Make sure the RTT buffers are drained
```

Comment:
This is necessary because sometimes when a breakpoint is hit, there are unprocessed data in the RTT buffers on the target. It is confusing for the user if not all data they expect to see are displayed back to them. During the halt, we can afford to build in some `thread::sleep` time to avoid cycling too fast.   However,  in code when the target is not halted, we poll for RTT data as fast as we can, to maximize throughput. 

The only part of the code that I can see that can be moved to a function is the single statement that starts with `if let Some(rtt_active_target) ...`, because in the second case we need to do a whole lot of other processing with the `core_data` handle.